airflow/models/dagrun.py

```diff
@@ -253,12 +253,14 @@ def __init__(
         dag_version: DagVersion | None = None,
         bundle_version: str | None = None,
     ):
+        # For manual runs where logical_date is None, ensure no data_interval is set.
+        if logical_date is None and data_interval is not None:
+            raise ValueError("data_interval must be None if logical_date is None")
+
         if data_interval is None:
-            # Legacy: Only happen for runs created prior to Airflow 2.2.
-            self.data_interval_start = self.data_interval_end = None
+            self.data_interval_start, self.data_interval_end = None, None
```

Comment:
We probably don't really need to change this line. Both are readable for me. not sure how others think 🤔

Reply:
changed it back.

Reply:
I have changed it to the following now:
```
if data_interval is None:
    # Legacy: Only happen for runs created prior to Airflow 2.2.
    self.data_interval_start = None
    self.data_interval_end = None
```

Reply:
@Lee-W I have changed it back to 
```
self.data_interval_start, self.data_interval_end = None, None
``` as discussed on slack

Reply:
changed it back. It was `,` instead of `=`